int32 data
string detect
float32 CO_data
float32 gas_data
